### DISCUSSION TITLE
Swiper is broken when the data length is less than the slidesPerView.

### DISCUSSION THREAD

**Discussion by olcaneristi** [nolimits4web/swiper#5365] [Q&A] [answered]
Hey there!

There is a strange issue we encountered in the carousel we created using Swiper. (React.js)

Carousel breaks down when data length is less than slidesPerView.

Example: When we've data with 4 items, "slidesPerView: 5" on 1024 breakpoint, the Carousel was broken on image/item width and Swiper can't calculate. 

- When we shrink/expand the window, DevTools shows the width as 1px.
- Image width is correct when data length is greater than slidesPerView at 576. (Data length: 4, slidesPerView: 3)

** Is there anyone who can help us with this problem? We want to know what we did wrong.

https://user-images.githubusercontent.com/45832621/149526836-3677eefe-5656-48d0-bffe-c8854e450ff7.mp4

Our breakpoints are like this:

```
const swiperParams: SwiperParams = {
  modules: [Navigation, Pagination, Scrollbar, A11y, Lazy],
  spaceBetween: 10,
  // resistance: false,
  navigation: {
    prevEl: '.gallery-carousel__slider-prev',
    nextEl: '.gallery-carousel__slider-next',
    disabledClass: 'gallery-carousel__slider--disabled',
  },
  breakpoints: {
    450: {
      slidesPerView: 2,
      slidesPerGroup: 2,
    },
    576: {
      slidesPerView: 3,
      slidesPerGroup: 3,
    },
    768: {
      slidesPerView: 5,
      slidesPerGroup: 5,
    },
    1024: {
      slidesPerView: 6,
      slidesPerGroup: 6,
    },
  },
  wrapperTag: 'ul',
  lazy: true,
};
```

Our Swiper layout is like this: 
```

<Swiper {...swiperParams} onPaginationUpdate={onPaginationUpdate}>
            {media.map(
              (reviewItem, index) =>
                reviewItem.uri && (
                  <SwiperSlide key={index} tag="li">
                    {({ isActive }) => (
                      <div className="gallery-carousel__slider__review-item" tabIndex={isActive ? 0 : -1}>
                        <a
                          href={'#'}
                          className="gallery-carousel__slider__review-item__link"
                          aria-labelledby={reviewItem.headline}
                        >
                          <img
                            src={reviewItem.uri.replace(/h_\d+/, 'h_175')}
                            alt=""
                            className="gallery-carousel__slider__review-item__image"
                          />
                        </a>
                      </div>
                    )}
                  </SwiperSlide>
                ),
            )}
</Swiper>
```

**Comment by olcaneristi** [comment, marked as answer]
ah, that's my fault. Swiper is broken cause of the flexbox in the main container. 

fixed.

**Reply by ivanjeremic** [reply]
Not really I have the same issue still, when the length of items is equal or higher than `slidesPerView` the slider breaks, does anyone know why?

**Reply by olcaneristi** [reply]
don't use flexbox (display:flex align-items:center, justify-content:center) on parent slider container. you can find the swiper problem with removing the classnames.

**Reply by ivanjeremic** [reply]
but isn't removing `class="swiper-wrapper"` breaking the slider?

**Reply by olcaneristi** [reply]
my swiper layout like this: 
![image](https://user-images.githubusercontent.com/45832621/149823308-34647917-97c1-4ed7-9d00-5e6abbf196cf.png)

I removed the SCSS elements in the div I marked and it was fixed. Swiper doesn't like flexbox on container elements I think.

**Reply by ivanjeremic** [reply]
I use React.js too but I don't know where the classes you marked come from, it seems to me this has nothing to do with the library all I use is the `Swiper` & `SwiperSlide` components from `swiper/react`. And I also checked my parent and I have no parent which could be a flexbox.

**Reply by olcaneristi** [reply]
yeah, these classes I marked are my custom layout elements. that was my problem. honestly, I don't understand what's causing your problem. I hope you can find a solution.

**Reply by ivanjeremic** [reply]
Yeah I hope so beauce flexbox isn't really my issue because I have no flexbox but still the problem with when length of my map is greater than or equal to `sliderPerView` it breaks.

**Reply by NeoHuncho** [reply]
did you find the solution? I'm having the same issue

**Reply by Hasan-Mir** [reply]
@olcaneristi Thank you sooooo much!
